projects/laji/src/app/+observation/download/observation-download.component.ts

```diff
@@ -74,6 +74,7 @@ export class ObservationDownloadComponent implements OnDestroy {
   requests: {[place: string]: RequestStatus} = {};
   requestStatus = RequestStatus;
   downloadLoading = false;
+  downloadProgressPercent?: number;
```

Comment:
I wonder if the logic of `downloadLoading` and `downloadProgressPercent` could be combined into one single variable by just removing `downloadLoading` and then a numeric value of `downloadProgressPercent` would imply that a download is loading. A single source of truth helps preventing bugs in the future and simplifies things.

Is there some specific reason for using `null` for `downloadProgressPercent`, instead of `undefined`? Now the value for `downloadProgressPercent` can be either `undefined` (the initial value), `null` or a numeric value. We should aim at using only either `undefined` or `null` for empty values in order to keep things consistent. We probably use `undefined` more often, and that's probably for the best since it eliminates cases where there are multiple possible values for nil values (non-initialized values are `undefined` anyway, so the solutions that make sense are either 1) use `undefined` or 2) use `null`, and initialize the variable with `null`. The 1st option is less work.)

Reply:
The variables `downloadLoading` and `downloadProgressPercent` are separate because only the GIS download has the progress percent and we don't want to show any percent if the percent is not available.

I replaced those `null` values with `undefined`

Reply:
Thanks! I added a few commits that made the `laji-download` & `laji-file-download` html templates to check for `undefined` instead of `null`. If it looks OK to you we can approve this

Reply:
I changed those to strict equalities (`!==`) since the variable can't be anything else than a number or undefined

Reply:
Good catch, thanks!